func/omp/default_shared.cpp

```diff
@@ -0,0 +1,72 @@
+#include <cstdio>
+#include <omp.h>
+
+#ifdef __wasm
+#include <faasm/shared_mem.h>
```

Comment:
I know we discussed this would potentially take place in a pre-processing step but could we hide the definition (or nulling-out) of `FAASM_REDUCE` in a header file?

Also, it is not clear to me why we check for the `__wasm` macro here and not in the other functions modified in this PR.

Reply:
This check doesn't actually need to be here so I'll remove it. However, this pattern is required for any code we want to compile natively, otherwise the Faasm macros will cause compiler errors.

I'm not sure which header we could put this in to hide it. We can't/ don't want to assume any Faasm-specific headers will be present in the native build, so the only way we could hide things is doing a fully transparent port of the OpenMP pragmas (so that native code would be completely unaware of Faasm).